src/plugins/dataflow-tool/components/ui/dataflow-program-topbar.tsx

```diff
@@ -1,137 +1,131 @@
-                  <option key={rt.text} value={rt.val} disabled={rt.disabled}>
-                    {rt.text}
-                  </option>
-                ))
-              }
+              { props.rateOptions.map((rate: ProgramDataRate) => (
+                <option key={rate.text} value={rate.val} disabled={rate.disabled}>
+                  {rate.text}
+                </option>
+              ))}
             </select>
           </div>
         </div>
       </div>
-      <button
-        className="program-state-button"
-        title="Run Program"
-        onClick={props.onRunProgramClick}
-        disabled={!props.isRunEnabled || props.readOnly}
-      >
-        <div className="icon run" />
-        <div className="text">Run</div>
-      </button>
     </div>
   );
 };
 
+interface RecordButtonProps {
+  readOnly: boolean;
+}
+
+const RecordButton: React.FC<RecordButtonProps> = (props: RecordButtonProps) => {
+  return (
+    <button
+      className="program-state-button"
+      title="Record"
+      onClick={() => { return null; }}
+      disabled={props.readOnly}
+    >
+      <div className="icon run" />
+      <div className="text">Record</div>
+    </button>
+  );
+};
+
 export const DataflowProgramTopbar = (props: TopbarProps) => {
-  const runTime = props.programRunTimes.find( (rt: ProgramRunTime) => rt.val === props.programDefaultRunTime );
-  const remainingHours = Math.floor(props.remainingTimeInSeconds / 3600);
-  const remainingMinutes = Math.floor((props.remainingTimeInSeconds - remainingHours * 3600) / 60);
-  const remainingSeconds = (props.remainingTimeInSeconds - (remainingHours * 3600) - (remainingMinutes * 60) ) % 60;
-  const completedTimeInSeconds = props.programDefaultRunTime - props.remainingTimeInSeconds;
-  const progressWidth = (kProgressWidth * completedTimeInSeconds / props.programDefaultRunTime);
-  const hoursString = String(remainingHours).padStart(2, "0");
-  const minutesString = String(remainingMinutes).padStart(2, "0");
-  const secondsString = String(remainingSeconds).padStart(2, "0");
   return (
     <div className="program-editor-topbar">
-      <div className="refresh">
+      <div className="topbar-left"></div>
+      <div className="topbar-center">
+        <RateSelectorComponent
+          rateOptions={props.programDataRates}
+          dataRate={props.dataRate}
+          onRateSelectClick={props.onRateSelectClick}
+          readOnly={props.readOnly}
+        />
+        <RecordButton readOnly={props.readOnly} />
+        <button
+          className="program-state-button"
+          title="Stop Program"
+          onClick={props.onStopProgramClick}
+          disabled={!props.runningProgram || !props.readOnly}
+        >
+          <div className="icon stop" />
+          <div className="text">Stop</div>
+        </button>
+      </div>
+      <div className="topbar-right">
+        {props.showRateUI && <span className={"rate-ui"}>{`${props.lastIntervalDuration}ms`}</span>}
```

Comment:
At the faster rates it can get hard to read the numbers. We may want to show a moving average or something in those situations, but that can be left as a problem for another day.